Task2/CsvEnumerable/CsvEnumerable/Program.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using CsvHelper;
+using System.IO;
+using System.Collections.Generic;
+using System.Linq;
+using CsvClassLibrary;
+
+namespace CsvEnumerable
+{
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            using (var sr = new StreamReader("info.csv"))
+            {
+                var reader = new CsvReader(sr);
+
+                IEnumerable<TestDataRecord> records = reader.GetRecords<TestDataRecord>();
```

Comment:
Let's complicate this task: currently you have implemented enumerator for simple list. Please, move records reading logic to enumerator. It means, that you read next portion of data only on `MoveNext` call and not in advance.